src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -0,0 +1,490 @@
+
+                // Setup the onError callback
+                _onError = new Action<JSObject>((errorEvt) =>
+                {
+                    errorEvt.Dispose();
+                });
+
+                // Attach the onError callback
+                _innerWebSocket.SetObjectProperty("onerror", _onError);
+
+                // Setup the onClose callback
+                _onClose = new Action<JSObject>((closeEvt) =>
+                {
+                    using (closeEvt)
+                    {
+                        _innerWebSocketCloseStatus = (WebSocketCloseStatus)closeEvt.GetObjectProperty("code");
+                        _innerWebSocketCloseStatusDescription = closeEvt.GetObjectProperty("reason")?.ToString();
+                        _receiveMessageQueue.Writer.TryWrite(new ReceivePayload(Array.Empty<byte>(), WebSocketMessageType.Close));
+                        NativeCleanup();
+                        if ((InternalState)_state == InternalState.Connecting)
+                        {
+                            tcsConnect.SetException(new WebSocketException(WebSocketError.NativeError));
+                        }
+                        else
+                        {
+                            _tcsClose?.SetResult();
+                        }
+                    }
+                });
+
+                // Attach the onClose callback
+                _innerWebSocket.SetObjectProperty("onclose", _onClose);
+
+                // Setup the onOpen callback
+                _onOpen = new Action<JSObject>((evt) =>
+                {
+                    using (evt)
+                    {
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            // Change internal _state to 'Connected' to enable the other methods
+                            if (Interlocked.CompareExchange(ref _state, (int)InternalState.Connected, (int)InternalState.Connecting) != (int)InternalState.Connecting)
+                            {
+                                // Aborted/Disposed during connect.
+                                tcsConnect.SetException(new ObjectDisposedException(GetType().FullName));
+                            }
+                            else
+                            {
+                                tcsConnect.SetResult();
+                            }
+                        }
+                        else
+                        {
+                            tcsConnect.SetCanceled(cancellationToken);
+                        }
+                    }
+                });
+
+                // Attach the onOpen callback
+                _innerWebSocket.SetObjectProperty("onopen", _onOpen);
+
+                // Setup the onMessage callback
+                _onMessage = new Action<JSObject>((messageEvent) =>
+                {
+                    // get the events "data"
+                    using (messageEvent)
```

Comment:
Could you extract this lambda into normal method?

Reply:
Addressed